### PR TITLE
feat: add typegen and docs for RawSlice, Bytes, and StdString

### LINKED ISSUE #1334: Update ABI typegen to support RawSlice, Bytes, and StdString
### Motivation

TS-SDK support was added for each of these types in Contracts, Scripts, and Predicates where possible (see exceptions in https://github.com/FuelLabs/fuels-ts/issues/1187).

The ABI typegen was not updated to utilize these newly supported types

See also:
- Vectors can be returned from Contracts https://github.com/FuelLabs/fuels-ts/issues/933
- Bytes can be sent and returned from Contracts https://github.com/FuelLabs/fuels-ts/issues/1188
- Vectors can be sent to Scripts https://github.com/FuelLabs/fuels-ts/pull/1046
- Bytes/RawSlices can be sent to Scripts https://github.com/FuelLabs/fuels-ts/issues/1274
- StdStrings https://github.com/FuelLabs/fuels-ts/issues/1276

### PR DESCRIPTION
**Pull request by camsjams**
Closes #1334

### CODE REVIEW

File: apps/docs/src/guide/types/bytes.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Bytes
+
+A dynamic array of byte values can be represented using the `Bytes` type, type used to represent raw bytes.
```
Comment (arboleya) [inline apps/docs/src/guide/types/bytes.md:3]:
```suggestion
A dynamic array of byte values can be represented using the `Bytes` type, which represents raw bytes.
```

File: apps/docs/src/guide/types/raw-slice.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# RawSlice
+
+A dynamic array of values can be represented using the `RawSlice` type. A raw slice can be a value reference or from a raw pointer.
```
Comment (arboleya) [inline apps/docs/src/guide/types/raw-slice.md:3]:
```suggestion
A dynamic array of values can be represented using the `RawSlice` type. A raw slice can be a value reference or a raw pointer.
```

**Review by arboleya — APPROVED** [review]
Great stuff!

**Review by danielbate — APPROVED** [review]

**Review by Torres-ssf — DISMISSED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Review by digorithm — APPROVED** [review]
:shipit:

**Pull request merged**